camel-idea-plugin/src/main/java/org/apache/camel/idea/annotator/CamelJSonPathAnnotator.java

```diff
@@ -0,0 +1,134 @@
+import org.apache.camel.idea.service.CamelService;
+import org.apache.camel.idea.util.CamelIdeaUtils;
+import org.apache.camel.idea.util.IdeaUtils;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Validate JSonPath expression and annotated the specific jsonpath expression to highlight the error in the editor
+ */
+public class CamelJSonPathAnnotator extends AbstractCamelAnnotator {
+
+    private static final Logger LOG = Logger.getInstance(CamelEndpointAnnotator.class);
+
+    @Override
+    boolean isEnabled() {
+        return ServiceManager.getService(CamelPreferenceService.class).isRealTimeJSonPathValidation();
+    }
+
+    /**
+     * Validate jsonpath expression. eg jsonpath("$.store.book[?(@.price < 10)]")
+     * if the expression is not valid a error annotation is created and highlight the invalid value.
+     */
+    void validateText(@NotNull PsiElement element, @NotNull AnnotationHolder holder, @NotNull String text) {
+
+        // only validate if the element is jsonpath element
+        if (getCamelIdeaUtils().isCamelJSonPathExpression(element)) {
+            CamelCatalog catalogService = ServiceManager.getService(element.getProject(), CamelCatalogService.class).get();
+            CamelService camelService = ServiceManager.getService(element.getProject(), CamelService.class);
+
+            // must have camel-json library
+            boolean jsonLib = camelService.containsLibrary("camel-jsonpath", false);
+            if (!jsonLib) {
+                camelService.showMissingJSonPathJarNotification(element.getProject());
+                return;
+            }
+
+            try {
+                // need to use the classloader that can load classes from the project
+                ClassLoader loader = camelService.getProjectClassloader();
+                if (loader != null) {
+                    LanguageValidationResult result;
+                    boolean predicate = getCamelIdeaUtils().isCameJSonPathExpressionUsedAsPredicate(element);
+                    if (predicate) {
+                        LOG.debug("Inspecting jsonpath predicate: " + text);
+                        result = catalogService.validateLanguagePredicate(loader, "jsonpath", text);
+                    } else {
+                        LOG.debug("Inspecting jsonpath expression: " + text);
+                        result = catalogService.validateLanguageExpression(loader, "jsonpath", text);
+                    }
+                    if (!result.isSuccess()) {
+                        String error = result.getShortError();
+                        if (error == null) {
+                            error = result.getError();
+                        }
+                        TextRange range = element.getTextRange();
+                        if (result.getIndex() > 0) {
+                            range = getAdjustedTextRange(element, range, text, result);
+
+                        }
+                        holder.createErrorAnnotation(range, error);
```

Comment:
consider split this up in small methods, like "validateLanguage" ,"displayErrorAnnoation" eg.